addons/mail/models/discuss/discuss_channel.py

```diff
@@ -512,10 +515,12 @@ def add_members(
                     payload["invited_by_user_id"] = self.env.user.id
                 member._bus_send("discuss.channel/joined", payload)
                 if post_joined_message:
+                    message_author = current_partner if current_partner else current_guest
+                    __, author_name = self.get_displayed_name(member=member, author=message_author)
+                    _ = self.env._
                     notification = (
-                        _("joined the channel")
-                        if member.is_self
-                        else _("invited %s to the channel", member._get_html_link(for_persona=True))
+                        _("%(user)s joined the channel", user=member._get_html_link(for_persona=True)) if member.is_self else
+                        _("%(user1)s invited %(user2)s to the channel", user1=author_name, user2=member._get_html_link(for_persona=True))
```

Comment:
While we are at it I think we should always use `_get_html_link`, even for user 1. No reason not being able to click on them.